api-lib/profileImageMutations.ts

```diff
@@ -0,0 +1,69 @@
+import { VercelRequest } from '@vercel/node';
+
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  uploadImageInput,
+} from '../src/lib/zod';
+
+import { gql } from './Gql';
+
+export const userAndImageData = (
+  req: VercelRequest
+): {
+  input: { image_data_base64: string };
+  hasuraProfileId: number;
+} => {
+  const {
+    input: { object: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    uploadImageInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+  return { input, hasuraProfileId: sessionVariables.hasuraProfileId };
+};
+
+export const profileUpdateImageMutation = (
+  hasuraProfileId: number,
+  _set: (fileName: string) => any
+) => {
+  // save the new image id in the db
+  return async (fileName: string): Promise<any> => {
+    const mutationResult = await gql.q('mutation')({
+      update_profiles_by_pk: [
+        {
+          _set: _set(fileName),
+          pk_columns: { id: hasuraProfileId },
+        },
+        {
+          id: true,
+        },
+      ],
+    });
+
+    if (mutationResult.update_profiles_by_pk) {
+      return {
+        id: mutationResult.update_profiles_by_pk.id,
+        profile: mutationResult.update_profiles_by_pk,
+      };
+    }
+    throw 'mutation was not successful';
+  };
+};
+
+export const profilePreviousImageQuery = async (
+  hasuraProfileId: number,
+  returning: () => any
+): Promise<{ avatar?: string; background?: string } | undefined> => {
+  // Figure out if there was a previous avatar, because we'll need to delete it
+  const { profiles_by_pk } = await gql.q('query')({
+    profiles_by_pk: [
+      {
+        id: hasuraProfileId,
+      },
+      returning(),
```

Comment:
```suggestion
      {background: true, avatar: true},
```

Reply:
this is a good easy change, i've pushed it